### PR TITLE
Add test case 1.1.2.1-convertValidSoapToJson

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
Purpose of this PR is to add test cases to product scenario

## Approach
Module 1.1.2-soap-to-json-using-messagetype-property has added. Relevant pom.xml files and properties have updated. 

## Test environment
This is tested locally in following environment.
JDK - 1.8
OS - macOS High Sierra

### CODE REVIEW

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/resources/testng.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <suite name="ESBTestSuite" parallel="false">
     <test name="ProxyService-Test" preserve-order="true" verbose="2" parallel="false">
         <classes>
-            <class name="org.wso2.carbon.esb.endpoint.test.ScenarioSampleTest"/>
+            <!--<class name="org.wso2.carbon.esb.endpoint.test.ScenarioSampleTest"/>-->
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/resources/testng.xml]: Shall we remove this line as not used?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        java.util.Collections.sort(requestFiles, Collator.getInstance());
+        java.util.Collections.sort(responseFiles, Collator.getInstance());
+
+        ArrayList<String> requestArray = new ArrayList();
+        ArrayList<String> responseArray = new ArrayList();
+
+
+        for (String file : requestFiles) {
+            File fileLocation = new File(getClass().getResource(File.separator + "source_files" + File.separator + "1.1.2.1" + File.separator + "request" + File.separator + file).getPath());
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java]: Please adhere to max character limit. Shall we break to multiple lines?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            String fileContent = getXmlContent(fileLocation);
+            requestArray.add(fileContent);
+        }
+
+        for (String file : responseFiles) {
+            String fileContent = readFile( getClass().getResource(File.separator + "source_files" + File.separator + "1.1.2.1" + File.separator + "response" + File.separator + file).getPath());
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java]: Please adhere to max character limit. Shall we break to multiple lines?

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java [hunk 4]
```diff
@@ -180,4 +194,33 @@ public Boolean call() throws Exception {
         };
     }
 
+    protected String getStringFromDocument(Document doc) throws TransformerException {
```
Comment (abeykoon) [inline product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: Shall we describe what this method does?

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java [hunk 5]
```diff
@@ -180,4 +194,33 @@ public Boolean call() throws Exception {
...
+        DOMSource domSource = new DOMSource(doc);
+        StringWriter writer = new StringWriter();
+        StreamResult result = new StreamResult(writer);
+        TransformerFactory tf = TransformerFactory.newInstance();
+        Transformer transformer = tf.newTransformer();
+        transformer.transform(domSource, result);
+        return writer.toString();
+    }
+
+    protected List<String> getListOfFiles(File filePath){
```
Comment (abeykoon) [inline product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: Shall we describe what this method does?

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java [hunk 6]
```diff
@@ -180,4 +194,33 @@ public Boolean call() throws Exception {
...
+
+        for (File file:listOfFiles) {
+            if (file.isFile()) {
+                fileNames.add(file.getName());
+            }
+        }
+        return fileNames;
+    }
+
+    protected String getXmlContent (File folderName) throws ParserConfigurationException, IOException, SAXException, TransformerException {
```
Comment (abeykoon) [inline product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: Shall we describe what this method does?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/resources/testng.xml (hunk 1 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/resources/testng.xml]: This is fixed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java (hunk 6 above)
Comment (dulanjalidilmi) [reply product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: This method was removed by [https://github.com/wso2/product-ei/pull/2853](https://github.com/wso2/product-ei/pull/2853)

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java (hunk 5 above)
Comment (dulanjalidilmi) [reply product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: Fixed by [https://github.com/wso2/product-ei/pull/3471](https://github.com/wso2/product-ei/pull/3471)

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java (hunk 4 above)
Comment (dulanjalidilmi) [reply product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: This method was removed by [https://github.com/wso2/product-ei/pull/2853](https://github.com/wso2/product-ei/pull/2853)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java (hunk 2 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java]: This was fixed by [https://github.com/wso2/product-ei/pull/2853](https://github.com/wso2/product-ei/pull/2853)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java (hunk 3 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.2-soap-to-json-using-messagetype-property/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonTransformationTest.java]: This was fixed by [https://github.com/wso2/product-ei/pull/2853](https://github.com/wso2/product-ei/pull/2853)

**Pull request merged**